tests/psa-client-server/psasim/test/start_server.sh

```diff
@@ -8,7 +8,7 @@ set -e
 # The server creates some local files when it starts up so we can wait for this
 # event as signal that the server is ready so that we can start client(s).
 function wait_for_server_startup() {
-    while [ $(find . -name "psa_notify_*" | wc -l) -eq 0 ]; do
+    while [ -z $(find /tmp -maxdepth 1 -name "psa_notify_*" -printf 1 -quit) ]; do
```

Comment:
The use of `-printf` means that this will no longer work on macOS

Reply:
Also putting files in `/tmp` is risky. We control build directories, so we should keep all files inside them.

I haven't read the whole code, but just looking at this line, it seems to me that if there are concurrent test runs on the same machine, they'll interfere with each other.

Reply:
A portable way to check if there are `psa_notify_*` files in the current directory:

```
files_exist () {
    [ $# -ge 2 ] || [ -e "$1" ] || [ -L "$1" ]
}
while files_exist 'psa_notify_*'; do ...
```

Reply:
> The use of `-printf` means that this will no longer work on macOS

ah okay

 

> A portable way to check if there are `psa_notify_*` files in the current directory:
> 
> ```
> files_exist () {
>     [ $# -ge 2 ] || [ -e "$1" ] || [ -L "$1" ]
> }
> while files_exist 'psa_notify_*'; do ...
> ```

I can do that. @tom-cosgrove-arm okay with that proposal?

Reply:
And regarding concurrent @gilles-peskine-arm yes, this isn't great. Ideally the server would output it's doorbell path when run, and the client would use that - no "hoping" to get the right one. But this right here is a quick fix.

I agree `/tmp` could be problematic, the question would be: would it be better to just check for each of two possible directories the files could be created in within the build tree, and use the first one found?

Reply:
@ronald-cron-arm err, yes, I am okay with `files_exist`, but see my comment above - I do agree with @gilles-peskine-arm that `/tmp` has its problems

Reply:
> would it be better to just check for each of two possible directories the files could be created in within the build tree, and use the first one found?

Sorry, I'm probably missing something because I haven't gone through the whole code and conversation. Why would we not know which directory the files are in?

Reply:
We used to put them in `./tests`, but that directory doesn't exist when we're in the framework (to a first approximation, I am multitasking severely ATM). So the suggestion: check if `./tests` is present. If so, use that, else use whatever directory is available when in the framework

Reply:
I will work on a better solution than /tmp.

Reply:
I've reverted the "Adapt psasim" commit and am proposing another solution to be able to run test suites located in `tests` and `tests/tf-psa-crypto`: starting two PSA servers, one "in" `tests`" and one "in" `tf-psa-crypto/tests`. @minosgalanakis @tom-cosgrove-arm please have a look, thanks.